### DISCUSSION TITLE
How to properly use ref with indiana drag scroll to set initial scroll position?

### DISCUSSION THREAD

**Discussion by dnh33** [Norserium/react-indiana-drag-scroll#62] [Q&A] [answered]
I've been trying to implement the initial scroll from https://codesandbox.io/s/react-indiana-drag-scroll-initial-scroll-getelement-99o6q?file=/index.js:317-382 

But haven't had much luck getting the ref to work it seems, or perhaps it is something with the render cycle. 

Here is my code to reference with the codesandbox:

 ```
 export default function ProjectCarousel() {
      const container = useRef(null);
    
      useEffect(() => {
        container.current.getElement().scrollTo(0, 240);
      }, []);
    
      return (
        <ScrollContainer className="scroll-container" ref={container}>
          {ProjectData.map(({ id, ...otherProps }) => (
            <ProjectImage key={id} {...otherProps} />
          ))}
        </ScrollContainer>
      );
    }
    
   ```

     

Thanks for any help in advance

**Comment by Norserium** [comment, marked as answer]
@dnh33, I can't answer on your question without additional details. As you can see, the code works in the sandbox, so it's not the library issue.

But I can assume, that the `ProjectData` (why its first letter is capital, by the way) is loaded after the `ProjectCarousel` mounted. Try to set `ProjectData` as the dependency of `useEffect` for the test purpose.

**Reply by dnh33** [reply]
Thank you for the quick reply!

Yes it's very odd, on a hot-reload after applying a value to the scrollTo it does seem to start the `scroll-container` at the specified value. But then when I reload the page (CTRL + R) it again starts the container at the top of the 1st image. 

The ProjectImage component is imported into Home.js

```
import ProjectCarousel from "../components/carousel/ProjectCarousel";

export default function Home() {
  return (
    <>
      <HomePage>
        <FlexedItems>
          <LeftFlexItem>
            <SocialBar />
            <Navbar />
            <Intro />
          </LeftFlexItem>
          <ProjectCarousel />
        </FlexedItems>
      </HomePage>
    </>
  );
}
```

Lmk if there is any other code you'd like to see :)

**Reply by Norserium** [reply]
@dnh33, did you try to add to the container some static blocks and test if the container scrolls on mount?

**Reply by dnh33** [reply]
> @dnh33, did you try to add to the container some static blocks and test if the container scrolls on mount?

Ah yes, the projectData, good you pointed that out, changed it all to projectData. But it loads into the component and is scroll-able.

**Reply by Norserium** [reply]
@dnh33, I don't get the last answer. 

Try to replace this:
```
{ProjectData.map(({ id, ...otherProps }) => (
     <ProjectImage key={id} {...otherProps} />
))}
```

By the static blocks styled blocks `<div className={'box'} />` with static width and height.

Also, I assume that there is nothing to scroll on mount, because your images are not loaded at this moment. Anyway, check the value of `scrollWidth` of `container.current.getElement()` before you scroll it.

**Reply by Norserium** [reply]
> Sometimes it doesn't register it still tho, would it be more effective to use the innerRef method?

No, it should not be more effective in this terms.

> On first initial page render, it seems to still be on the top of the 1st image, but if I navigate to /contact and back, it render with the scroll position

Perhaps, because image were cached.

**Reply by dnh33** [reply]
Thanks a bunch - that was it 👍🏻

It was the static height that was required applied on the div around the img tag

**Reply by dnh33** [reply]
Perhaps you could also answer how I can make the container scrollable with the mouse wheel on desktop after the website has been built. It seems to work in development mode only.

**Reply by Norserium** [reply]
@dh33, the temporary solution is the setting for the container element the following style:
```
.container {
    overflow: auto !important;
}
```

I should figure out something better, but that's what we got so far.

**Reply by dnh33** [reply]
that's perfect 👍🏻
